apps/site/src/pages/api/blocks/index.api.ts

```diff
@@ -23,12 +24,24 @@ export type ApiBlockSearchResponse = {
 };
 
 export default createApiKeyRequiredHandler<null, ApiBlockSearchResponse>()
-  .use(queryValidator("author").isString().toLowerCase())
-  .use(queryValidator("license").isString().toLowerCase())
-  .use(queryValidator("name").isString().toLowerCase())
-  .use(queryValidator("q").isString().toLowerCase())
-  .use(queryValidator("json").isJSON())
+  .use(queryValidator("author").optional().isString().toLowerCase())
+  .use(queryValidator("license").optional().isString().toLowerCase())
+  .use(queryValidator("name").optional().isString().toLowerCase())
+  .use(queryValidator("q").optional().isString().toLowerCase())
+  .use(queryValidator("json").optional().isJSON())
   .get(async (req, res) => {
+    const errors = validationResult(req);
+    if (!errors.isEmpty()) {
+      return res.status(400).json(
+        formatErrors(
+          ...errors.array().map((validationError) => ({
+            message: validationError.msg,
+            ...validationError,
+          })),
+        ),
+      );
+    }
```

Comment:
@CiaranMn have added the change we discussed in standup, but noticed that we aren't throwing any of the validation errors in the `/api/blocks` method when they occur. Have added the necessary handler here, and marked the query params as optional (as to my understanding they all should be). Here's an example of what a validation error would look like if the `json` parameter isn't JSON:
<img width="1086" alt="image" src="https://user-images.githubusercontent.com/42802102/223704462-2e2b6d11-5c78-4ce2-ba5c-26d0c4fab0e2.png">